airbyte-db/lib/src/main/java/io/airbyte/db/check/impl/ConfigsDatabaseAvailabilityCheck.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.db.check.impl;
+
+import io.airbyte.db.check.DatabaseAvailabilityCheck;
+import io.airbyte.db.instance.configs.ConfigsDatabaseInstance;
+import java.util.Optional;
+import org.jooq.DSLContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of the {@link DatabaseAvailabilityCheck} for the Configurations database.
+ */
+public class ConfigsDatabaseAvailabilityCheck implements DatabaseAvailabilityCheck {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ConfigsDatabaseAvailabilityCheck.class);
+
+  // TODO inject via dependency injection framework
+  private final DSLContext dslContext;
+
+  // TODO inject via dependency injection framework
+  private final long timeoutMs;
+
+  public ConfigsDatabaseAvailabilityCheck(final DSLContext dslContext, final long timeoutMs) {
```

Comment:
nit: we can use @AllArgConstructor to generate this constructor

Reply:
@benmoriceau This constructor is most likely temporary and will be removed when we add dependency injection.  At that point, we can just annotate the member fields with the injection information and can remove the constructor entirely.